src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -311,21 +310,22 @@ object Task {
     private[this] def hasStartedRunning: Boolean = status.startedAt.isDefined
 
     override def update(op: TaskUpdateOperation): TaskUpdateEffect = op match {
-      case TaskUpdateOperation.MesosUpdate(newStatus: Terminal, mesosStatus) =>
-        val updated = copy(status = status.copy(
-          mesosStatus = Some(mesosStatus),
-          taskStatus = newStatus))
-        TaskUpdateEffect.Expunge(updated)
-
-      case TaskUpdateOperation.MesosUpdate(InstanceStatus.Running, mesosStatus) if !hasStartedRunning =>
+      case TaskUpdateOperation.MesosUpdate(InstanceStatus.Running, mesosStatus, now) if !hasStartedRunning =>
         val updatedTask = copy(status = status.copy(
           mesosStatus = Some(mesosStatus),
           taskStatus = InstanceStatus.Running,
-          startedAt = Some(Timestamp(mesosStatus.getTimestamp.seconds.toMillis))
+          startedAt = Some(now)
         ))
         TaskUpdateEffect.Update(newState = updatedTask)
 
-      case TaskUpdateOperation.MesosUpdate(newStatus, mesosStatus) =>
+      // The Terminal extractor applies specific logic e.g. when an Unreachable task becomes Gone
+      case TaskUpdateOperation.MesosUpdate(newStatus: Terminal, mesosStatus, _) =>
+        val updated = copy(status = status.copy(
+          mesosStatus = Some(mesosStatus),
+          taskStatus = newStatus))
+        TaskUpdateEffect.Update(updated)
```

Comment:
The effect was expunge before the change. I guess we will update always on the level on tasks and expunge on the level of instances? (do we need TaskUpdateEffect.Expunge?)